openstack/clientconfig/requests.go

```diff
@@ -155,8 +174,39 @@ func GetCloudFromYAML(opts *ClientOpts) (*Cloud, error) {
 		if !ok {
 			return nil, fmt.Errorf("cloud %s does not exist in clouds-public.yaml", profileName)
 		}
+		cloud, err = mergeClouds(cloud, publicCloud)
+		if err != nil {
+			return nil, fmt.Errorf("Could not merge information from clouds.yaml and clouds-public.yaml for cloud %s", profileName)
+		}
+	}
 
-		updateAuthInfo(cloud, &publicCloud)
+	secureClouds, err := LoadSecureCloudsYAML()
+	if err != nil {
+		return nil, fmt.Errorf("unable to load secure.yaml: %s", err)
+	}
+
+	if secureClouds != nil {
```

Comment:
I might be incorrect, but can this block of code be rewritten as:

```go
// If no entry was found in clouds.yaml, no cloud name was specified,
// and only one secureCloud entry exists, use that as the cloud entry.
if !cloudIsInCloudsYaml && cloudName == "" && len(secureClouds) == 1 {
  for _, v := range secureClouds {
    cloud = &v
  }
}

secureCloud, ok := secureClouds[cloudName]
if !ok && cloud == nil {
  // cloud == nil serves two purposes here:
  // if no entry in clouds.yaml was found and
  // if a single-entry secureCloud wasn't used.
  // At this point, no entry could be determined at all.
  return nil, fmt.Errorf("Could not find cloud %s", cloudName)
}

// If secureCloud has content and it differs from the cloud entry,
// merge the two together.
if !reflect.DeepEqual((Cloud{}), secureCloud) && !reflect.DeepEqual(cloud, secureCloud) {
  cloud, err = mergeClouds(secureCloud, cloud)
  if err != nil {
    return nil, fmt.Errorf("unable to merge information from clouds.yaml and secure.yaml")
  }
}
```

Reply:
That order would indeed make more sense I think. I'll change it.
One thing that I just noticed: If clouds.yaml only has a single entry, no cloudName is specified and secure.yaml also only has a single entry, should the entries in clouds.yaml and secure.yaml be merged? Because at the moment, they won't be.

Reply:
If you can, confirm behavior with the `openstack` tool and let's match what it does. I know when there was 1 x clouds.yaml entry and 1 x vendor entry, I still had to specify a cloud/profile.